kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -470,7 +500,12 @@ def run_cfg_group(tests: list[str]) -> dict[str, bool]:
         raise ValueError(f'Running setUp method failed for {len(failed)} contracts: {failed}')
 
     _LOGGER.info(f'Running test functions in parallel: {tests}')
-    return run_cfg_group(tests)
+    results = run_cfg_group(tests)
+
+    for method in test_methods:
+        method.update_digest(foundry.out / 'digest')
+
+    return results
```

Comment:
Why are we updating the digests _after_ running the proofs? Could we do it much earlier, after we've initialized the KCFGs? For example, we could do it in `run_cfg_group` right after we do `write_proof()`?

That way, if we kill a proof partway through, we don't lose the up-to-date digest (which should be the same if the solidity code itself didn't change).

Does this make sense?

Reply:
Yes, that makes sense. I was thinking of it in terms of the proof being up to/out of date, but yes, if it's stopped part of the way through it will currently be reinitialized.

Reply:
Is there any reason we can't just update each digest right after we check it? Putting it after `write_proof()` would result in it being called a bunch of times unnecessarily, I think, since that is in `advance_proof()`.

Reply:
Yep, makes sense!